### PR TITLE
milestone 2, scaling func and test

### PR DESCRIPTION
**Pull request by ribner**

### CODE REVIEW

File: nursepy/preproc.py [hunk 1]
```diff
@@ -31,17 +31,38 @@ def preproc(X_train, X_test=None, auto=False, OHE=[], standard_scale=[], robust_
...
+    }
+    # check arg types
+    for arg_key in args.keys():
+        argument_type = type(args[arg_key])
+        if argument_type not in arg_types[arg_key]:
+            raise ValueError(
+                f"Arg type error. {arg_key} needs to be one of {arg_types[arg_key]}. You passed in type of: {argument_type}")
 
+    X_train = X_train.copy()
+    if (X_test is not None):
```
Comment (merveshin) [inline nursepy/preproc.py]:
```suggestion
    if X_test is not None:
```

File: nursepy/preproc.py [hunk 2]
```diff
@@ -31,17 +31,38 @@ def preproc(X_train, X_test=None, auto=False, OHE=[], standard_scale=[], robust_
...
+        X_test = X_test.copy()
     # automatically choose which columns to scale and encode
     if (auto == True):
```
Comment (merveshin) [inline nursepy/preproc.py]:
```suggestion
    if auto :
```

File: nursepy/preproc.py [hunk 3]
```diff
@@ -31,17 +31,38 @@ def preproc(X_train, X_test=None, auto=False, OHE=[], standard_scale=[], robust_
...
-        for element in [OHE, standard_scale, robust_scale, numerical_impute, categegorical_impute]:
+        for element in [OHE, standard_scale, robust_scale, numerical_impute, categorical_impute]:
             if(len(element) > 0):
```
Comment (merveshin) [inline nursepy/preproc.py]:
```suggestion
            if len(element) > 0 :
```

File: nursepy/preproc.py [hunk 4]
```diff
@@ -31,17 +31,38 @@ def preproc(X_train, X_test=None, auto=False, OHE=[], standard_scale=[], robust_
...
                 raise Exception(
                     f'You cannot manually set {element} when setting auto to true')
         label_keys = label_encode.keys()
         if (len(label_keys) > 0):
```
Comment (merveshin) [inline nursepy/preproc.py]:
```suggestion
        if len(label_keys) > 0 :
```

**Review by evhend — COMMENTED** [review]
No apparent issues with code - well done

**Review by merveshin — CHANGES_REQUESTED** [review]
A few minor suggestions to polish your code and I think renaming the test file would be better. Since we are writing tests for each function in a separate file, making the file name sth like `test_preproc.py` would be more clear and be better to not get any conflict issues I guess.

**Review by merveshin — APPROVED** [review]

**Review by merveshin — APPROVED** [review]

**Pull request merged**